### ISSUE TITLE
Screen reader can't edit typed text in blocks

### ISSUE THREAD

**Issue by rkingett** [WordPress/gutenberg#19124]
**Describe the bug**
When using the latest NVDA screen reader, and the latest version of FireFox, text can't be edited after it is typed in any text block. Markdown, paragraph, and similar. After composing a markdown block using NVDA in focus mode, the screen reader will only read the first character or the last character but it will not read any other characters when trying to arrow through.

**To reproduce**
The below requires NVDA:
1. Create a markdown block with NVDA in focus mode. Forms mode. '...'
2. After you have written more than one character, press the left and right keys to try to navigate within the text you just typed.
3. don't navigate away from the block after typing.

**Expected behavior**
I expected the blocks to let me edit and review the text as if it were a notepad document.

- OS: Windows 10, 1903.
 - Browser [Firefox 71.0 (32-bit) 71.0 (32-bit)

**Additional context**
I am using the latest version of the editor used on wordpress.com. Not wordpress.org. This problem does not happen when using HTML view.

**Comment by MarcoZehe** [comment]
Note that I cannot reproduce this. I wrote [this introduction to Gutenberg](https://marcozehe.de/2019/12/13/a-quick-introduction-to-using-gutenberg/) using WordPress.com and the version of Gutenberg it comes with, with NVDA and Firefox. I haven't specifically tried the markdown block, but other blocks such as paragraph, heading, etc., work fine for me. Also on Windows 10 1903.

**Comment by rkingett** [comment]
I tried it again this morning and, it seems to happen all the time when writing in a markdown block, without first leaving the block. The paragraph block does not have this bug happen.

**Comment by MarcoZehe** [comment]
I can definitely reproduce this with the markdown block, but not with a paragraph block. I am experimenting with a wild idea, but this is probably a manifestation of an obscure Firefox bug that sometimes hits when a textarea gets un-hidden and focus at the same time.

**Comment by tellthemachines** [comment]
The markdown block in WordPress.com comes from [the Jetpack repo](https://github.com/Automattic/jetpack/tree/master/extensions/blocks/markdown), not from Gutenberg. If this issue can't be reproduced with any Gutenberg block, perhaps it is Jetpack-specific.

**Comment by MarcoZehe** [comment]
It is reproducible with the Code block, too.

**Comment by tellthemachines** [comment]
> It is reproducible with the Code block, too.

No worries, in that case it's definitely on our side.

**Comment by rkingett** [comment]
It happens in the code block as well. I tried the paragraph multiple times to see if it would happen, using Chrome, and Edge BETA, and it happened twice out of 10. On Chrome and chromium Edge.

**Comment by enriquesanchez** [comment]
I tested on Firefox with NVDA and I was able to reproduce this, but only on the markdown block. The cursor moves left and right normally, but NVDA only announces the first letter of the sentence.


I also tested on Safari with VoiceOver and was not able to reproduce.

**Comment by afercia** [comment]
I see the Markdown block by Jetpack uses a role=group. That might explain why it doesn't work well. Also, when editing, a textarea element is dynamically injected within the block. I doubt this behavior plays well with screen readers.

For the blocks provided by core, we established long time ago that all the `contenteditable` elements of the blocks need to have a role=textbox with an aria-multiline=true so that they're perceived and behave like textarea elements.

I see the Code block uses a nested structure instead, not sure why:
- the first focusable element is a `<pre>` element with aria-label="Block: Code" role="group" tabindex="0"
- nested within the above element, there's a contenteditable element, which is a `<code>` element with role="textbox" aria-multiline="" aria-label="Code"

Also, the aria-multiline attribute appears to have an empty value for all the block types, which isn't correct. This seems a regression to me. It used to have a `true` value.
